nav2_util/include/nav2_util/odometry_utils.hpp

```diff
@@ -68,13 +68,44 @@ class OdomSmoother
    * @brief Get twist msg from smoother
    * @return twist Twist msg
    */
-  inline geometry_msgs::msg::Twist getTwist() {return vel_smooth_.twist;}
+  inline geometry_msgs::msg::Twist getTwist()
+  {
+    std::lock_guard<std::mutex> lock(odom_mutex_);
```

Comment:
that's pretty important right? `vel_smooth_` is not atomically updated

Reply:
Yeah - this is pretty old, back in the Intel + Steve days, so I think there was some R&D thinking and not production thinking back then